packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -564,11 +583,8 @@ Future<void> buildGradleAar({
       'Local Maven repo: ${localEngineRepo.path}'
     );
     command.add('-Plocal-engine-repo=${localEngineRepo.path}');
-    command.add('-Plocal-engine-build-mode=${androidBuildInfo.buildInfo.modeName}');
+    command.add('-Plocal-engine-build-mode=${buildInfo.modeName}');
     command.add('-Plocal-engine-out=${localEngineArtifacts.engineOutPath}');
-    if (androidBuildInfo.buildInfo.treeShakeIcons) {
-      command.add('-Pfont-subset=true');
-    }
```

Comment:
This is in the wrong spot, I moved it out of the local engine check.

Reply:
@dnfield Can you confirm you didn't intentionally want the tree shake check to only happen with a local engine?

Reply:
Correct - it should be able to work with both. I probably only tested with a local engine.